test/datastore/query.js

```diff
@@ -25,116 +25,333 @@ var Query = require('../../lib/datastore/query.js');
+      query = query
+        .filter('count <=', 5);
+
+      assert.equal(query.filters.length, 1);
+    });
+
+    it('should return a new query', function() {
+      var query = new Query(['kind1']);
+      var newQuery = query.filter('count <', 5);
+
+      assert.notDeepEqual(query.filters, newQuery.filters);
+    });
+
   });
 
-  it('should support single field selection by field name', function() {
-    var query = new Query(['kind1'])
-        .select('name');
-    assert.equal(query.selectVal[0], 'name');
+  describe('hasAncestor', function() {
+
+    it('should support ancestor filtering', function() {
+      var query = new Query(['kind1'])
+          .hasAncestor(['kind2', 123]);
+
+      assert.equal(query.filters[0].name, '__key__');
+      assert.equal(query.filters[0].op, 'HAS_ANCESTOR');
+      assert.deepEqual(query.filters[0].val, ['kind2', 123]);
+    });
+
+    it('should return a new query', function() {
+      var query = new Query(['kind1']);
+      var newQuery = query.hasAncestor(['kind2', 123]);
+
+      assert.notDeepEqual(query.filters, newQuery.filters);
+    });
+
   });
 
-  it('should support ancestor filtering', function() {
-    var query = new Query(['kind1'])
-        .hasAncestor(['kind2', 123]);
-    assert.equal(query.filters[0].name, '__key__');
-    assert.equal(query.filters[0].op, 'HAS_ANCESTOR');
-    assert.deepEqual(query.filters[0].val, ['kind2', 123]);
+  describe('order', function() {
+
+    it('should default ordering to ascending', function() {
+      var query = new Query(['kind1'])
+        .order('name');
+
+      assert.equal(query.orders[0].name, 'name');
+      assert.equal(query.orders[0].sign, '+');
+    });
+
+    it('should support ascending order', function() {
+      var query = new Query(['kind1'])
+          .order('+name');
+
+      assert.equal(query.orders[0].name, 'name');
+      assert.equal(query.orders[0].sign, '+');
+    });
+
+    it('should support descending order', function() {
+      var query = new Query(['kind1'])
+          .order('-count');
+
+      assert.equal(query.orders[0].name, 'count');
+      assert.equal(query.orders[0].sign, '-');
+    });
+
+    it('should support both ascending and descending', function() {
+      var query = new Query(['kind1'])
+          .order('+name')
+          .order('-count');
+
+      assert.equal(query.orders[0].name, 'name');
+      assert.equal(query.orders[0].sign, '+');
+      assert.equal(query.orders[1].name, 'count');
+      assert.equal(query.orders[1].sign, '-');
+    });
+
+    it('should create the orders property if it is a falsy value', function() {
+      var query = new Query(['kind1']);
+
+      query.orders = false;
+      query = query.order('size');
+
+      assert.equal(query.orders.length, 1);
+    });
+
+    it('should return a new query', function() {
+      var query = new Query(['kind1']);
+      var newQuery = query.order('count');
+
+      assert.notDeepEqual(query.orders, newQuery.orders);
```

Comment:
We could, my primary concern is that somewhere down the line a `return` is forgotten and `newQuery` becomes `null`. 

```javascript
assert.notStrictEqual(query, null);
```